core/src/com/projectkorra/projectkorra/PKListener.java

```diff
@@ -521,14 +471,20 @@ public void onEntityChangeBlockEvent(final EntityChangeBlockEvent event) {
 	@EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
 	public void onEntityCombust(final EntityCombustEvent event) {
 		final Entity entity = event.getEntity();
+		if (!(entity instanceof LivingEntity)) {
+			return;
+		}
 		final Block block = entity.getLocation().getBlock();
-		if (FireAbility.getSourcePlayers().containsKey(block) && entity instanceof LivingEntity) {
-			new FireDamageTimer(entity, FireAbility.getSourcePlayers().get(block));
+		final Player source = FireAbility.getSourcePlayers().get(block);
+		if (source != null) {
+			new FireDamageTimer(entity, source, null, false);
 		}
 	}
 
 	@EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
 	public void onEntityDamageByBlock(final EntityDamageByBlockEvent event) {
+		// TODO: Review if this listener is needed at all, EntityDamageByBlockEvent is also caught by onEntityDamageEvent and that method
+		// has all of the below logic in it already, which makes me think this happens twice now, not that it matters but its wasted tick time
```

Comment:
Could someone look over this specifically, don't want to touch either of these two methods till im sure on this